extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,886 @@
+      repo,
+      ref: sha,
+    });
+
+    // Create a temporary directory for downloading
+    const archivePath = join(
+      tmpDir.name,
+      `source-${owner}-${repo}-${Date.now()}.zip`,
+    );
+
+    // Set timeout
+    const {
+      signal,
+      onData,
+      dispose: disposeTimeout,
+    } = createTimeoutSignal(downloadTimeout());
+
+    // Fetch the url
+    let response: Response;
+    try {
+      response = await fetch(url, {
+        headers: {
+          Accept: "application/zip",
+        },
+        signal,
+      });
+    } catch (e) {
+      disposeTimeout();
+
+      if (e instanceof DOMException && e.name === "AbortError") {
+        const thrownError = new Error("The request timed out.");
+        thrownError.stack = e.stack;
+        throw thrownError;
+      }
+      throw new Error(
+        `Error fetching source root. Reason: ${getErrorMessage(e)}`,
+      );
+    }
+
+    // Download the source root from the response body
+    const body = response.body;
+    if (!body) {
+      throw new Error("No response body found");
+    }
+
+    const archiveFileStream = createWriteStream(archivePath);
+
+    const contentLength = response.headers.get("content-length");
+    const totalNumBytes = contentLength
+      ? parseInt(contentLength, 10)
+      : undefined;
+
+    const reportProgress = reportStreamProgress(
+      "Downloading source root...",
+      totalNumBytes,
+      progressCallback,
+    );
+
+    try {
+      const reader = body.getReader();
+      for (;;) {
+        const { done, value } = await reader.read();
+        if (done) {
+          break;
+        }
+
+        onData();
+        reportProgress(value?.length ?? 0);
+
+        await new Promise((resolve, reject) => {
+          archiveFileStream.write(value, (err) => {
+            if (err) {
+              reject(err);
+            }
+            resolve(undefined);
+          });
+        });
+      }
+
+      await new Promise((resolve, reject) => {
+        archiveFileStream.close((err) => {
+          if (err) {
+            reject(err);
+          }
+          resolve(undefined);
+        });
+      });
+    } catch (e) {
+      // Close and remove the file if an error occurs
+      archiveFileStream.close(() => {
+        void remove(archivePath);
+      });
+
+      if (e instanceof DOMException && e.name === "AbortError") {
+        const thrownError = new Error("The download timed out.");
+        thrownError.stack = e.stack;
+        throw thrownError;
+      }
+
+      throw new Error(
+        `Error downloading source root. Reason: ${getErrorMessage(e)}`,
+      );
+    } finally {
+      disposeTimeout();
+    }
+
+    void extLogger.log(`Download complete, extracting source...`);
+
+    // Extract the downloaded zip file
+    await unzipToDirectoryConcurrently(
+      archivePath,
+      outputPath,
+      progressCallback
+        ? reportUnzipProgress(`Unzipping source root...`, progressCallback)
+        : undefined,
```

Comment:
I think the `progressCallback` is always defined here:

```suggestion
      reportUnzipProgress(`Unzipping source root...`, progressCallback),
```

Reply:
Ah, thanks, copy-paste issue. Updated in https://github.com/github/vscode-codeql/pull/4065/commits/0e6aa813734acc575b118799fde32794325fddfd.